androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -38,8 +38,12 @@ fun SurveyDetailScreen(
     var imageScale by remember { mutableStateOf(InitialImageScale) }
     val coroutineScope = rememberCoroutineScope()
     val surveyUiModel = homeViewState.surveys.find { it.id == surveyId }
-    val surveyWithoutIntro = surveyDetailViewState.survey?.run {
-        copy(questions = questions?.filter { it.displayType != INTRO })
+    var questionsWithoutIntro by remember(surveyDetailViewState.survey) {
+        mutableStateOf(
+            surveyDetailViewState.survey?.run {
+                copy(questions = questions?.filter { it.displayType != INTRO })
+            }?.toSurveyUiModel()?.questionUiModels
+        )
```

Comment:
This kind of filtering needs to be in the `ViewModel` instead. Please remove the data related logics from the views. 🙏

Reply:
As I see it, this is a UI logic — VM holds the data, but the UI uses it this way. You're right that it's better to keep the presentation logic in VM, but the same logic is used by iOS as well. Any suggestions?

Reply:
Not sure how we can approach this multi-platform issue since I don't have much knowledge of KMM. Can we not share the filtering logic in VM?

Reply:
@kaungkhantsoe we can, but it won't be used by iOS. Keeping both sides consistent in terms of logic would be nice IMO. Let me check again if there's something we can do about it.